crates/types/src/traits/election.rs

```diff
@@ -775,6 +777,15 @@ pub trait QuorumExchangeType<TYPES: NodeType, LEAF: LeafType<NodeType = TYPES>,
         leaf_commitment: &Commitment<LEAF>,
     ) -> EncodedSignature;
 
+    /// Sign a block commitment.
+    ///
+    /// TODO GG: `sign_block_commitment` and `sign_validating_or_commitment_proposal` should be the same function.
+    /// `Commitment<LEAF>` should equal `Commitment<TYPES::BlockType>`.
```

Comment:
👍

Reply:
Per above discussion, leaf commitments add metadata to a block commitment, so these two commitments should *not* be equal.